generate/generate.go

```diff
@@ -59,7 +61,9 @@ func Generate(dir string, patterns ...string) error {
 	for _, pkg := range pkgs {
 		g.gunkPkgs[pkg.PkgPath] = pkg
 		for i, name := range pkg.GunkNames {
-			g.origPaths[name] = pkg.GunkFiles[i]
+			// Save only the directory for each file path.
+			dir, _ := filepath.Split(name)
+			g.origPaths[dir], _ = filepath.Split(pkg.GunkFiles[i])
```

Comment:
I'm not sure why this change is necessary - you seem to do the same `filepath.Split` when querying and using `origPaths`, so it seems to me like the old way would be simpler

Reply:
The old way includes the filename, which makes it kind of hard to match when getting the modifed output filenames from a protoc-gen-* generator. Since this is only used to know where the package path points to on disk, it is easier to just check the "directory" path since that won't change between `protoc-gen-*` runs, rather than changing `github.com/gunk/gunk/examples/utils/echo.pb.gw.go` to `github.com/gunk/gunk/examples/utils/echo.gunk` and then doing the lookup for the on-disk location.

Reply:
Ah gotcha - you're probably right.

What we should do in the long run is use the dir where the gunk file is as the default output path, and allowing the gunk config to override that. For example, in brankas-proto, `pb/v2/*` is placed in `v2/*`.